collector/processor/concurrentbatchprocessor/batch_processor.go

```diff
@@ -467,11 +474,13 @@ func (mb *multiShardBatcher) consume(ctx context.Context, data any) error {
 		select {
 		case newErr := <-respCh:
 			// nil response might be wrapped as an error.
-			if errors.Unwrap(newErr) != nil {
+			unwrap := newErr.(countedError)
+			if unwrap.err != nil {
 				err = multierr.Append(err, newErr)
 			}
 
-			if isPartialError(newErr) {
+			item.count -= unwrap.count
+			if item.count > 0 {
```

Comment:
nit -- in one place you have `item.count != 0` and the other `item.count > 0`. please make the two statements have one style.

(since I'm the one who split this code in the first place, I'm the one I want to blame when I say this: I wish there were only one copy of this logic, but I don't want to slow you down by saying so!)

Reply:
Thanks for catching that! I now moved the common code into a new shard method called `shard.consumeAndWait`